overlord/devicestate/devicestate_install_mode_test.go

```diff
@@ -726,3 +726,71 @@ func (s *deviceMgrInstallModeSuite) TestInstallModeWritesModel(c *C) {
 
 	c.Check(filepath.Join(boot.InitramfsUbuntuBootDir, "device/model"), testutil.FileEquals, buf.String())
 }
+
+func (s *deviceMgrInstallModeSuite) testInstallGadgetNoSave(c *C) {
+	err := ioutil.WriteFile(filepath.Join(dirs.GlobalRootDir, "/var/lib/snapd/modeenv"),
+		[]byte("mode=install\n"), 0644)
+	c.Assert(err, IsNil)
+
+	s.state.Lock()
+	s.makeMockInstalledPcGadget(c, "dangerous", "")
+	info, err := snapstate.CurrentInfo(s.state, "pc")
+	c.Logf("mount: %v", info.MountDir())
+	c.Assert(err, IsNil)
```

Comment:
super minor nitpick, but what if `info` is nil? won't this test panic? maybe this is better as

```suggestion
	c.Assert(err, IsNil)
	c.Logf("mount: %v", info.MountDir())
```
?

not a blocker either way feel free to leave as-is